test/metabase/upload_test.clj

```diff
@@ -1230,10 +1231,18 @@
   (try
     (mt/with-temporary-setting-values [uploads-enabled uploads-enabled]
       (mt/with-current-user user-id
-        (let [table-id (or table-id (:id (create-upload-table!)))]
-          (t2/update! :model/Table table-id {:is_upload is-upload})
-          (append-csv! {:table-id table-id
-                        :file     file}))))
+        (mt/with-model-cleanup [:model/Table]
+          (let [new-table (when (nil? table-id)
+                            (create-upload-table!))
+                table-id (or table-id (:id new-table))]
+            (t2/update! :model/Table table-id {:is_upload is-upload})
+            (try (append-csv! {:table-id table-id
+                               :file     file})
+                 (finally
+                   (when new-table
+                     (driver/drop-table! driver/*driver*
+                                         (mt/id)
+                                         (#'upload/table-identifier new-table)))))))))
     (finally
       (io/delete-file file))))
```

Comment:
This is a preexisting smell, but it's a footgun that the method will delete the file even if it did not create it.